service/storage_service.cc

```diff
@@ -4359,6 +4384,44 @@ future<> storage_service::raft_rebuild(sstring source_dc) {
     });
 }
 
+future<> storage_service::raft_check_and_repair_cdc_streams() {
+    std::optional<cdc::generation_id_v2> curr_gen;
+
+    while (true) {
+        slogger.info("raft topology: request check_and_repair_cdc_streams, refreshing topology");
+        auto guard = co_await _group0->client().start_operation(&_abort_source);
+        auto curr_req = _topology_state_machine._topology.global_request;
+        if (curr_req && *curr_req != global_topology_request::new_cdc_generation) {
+            // FIXME: replace this with a queue
```

Comment:
One reason I believe we need the queue is that I think topology state transitions could be handled by manipulating the queue, similarly to how reverse polish notation works.
For example, adding a node could expand to:
- pop adding a node task from the queue
- push change raft config, handle state transitions, insert cdc generations into the queue.

Another advantage of having a queue is being able to quickly register multiple topology requests yet work them off one at a time. 
This is already partly done by register the global topology state transitions such as "write both read new" from within an existing topology operation.